athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeMetadataHandler.java

```diff
@@ -178,6 +181,48 @@ public Schema getPartitionSchema(final String catalogName)
                 .addField(BLOCK_PARTITION_COLUMN_NAME, Types.MinorType.VARCHAR.getType());
         return schemaBuilder.build();
     }
+
+    private String getPrimaryKey(TableName tableName) throws Exception 
```

Comment:
I'd personally prefer to use Optional instead of returning am empty string. It's easier to check for.